src/lucky_support/inflector/inflections.cr

```diff
@@ -0,0 +1,142 @@
+        @uncountables = Uncountables.new
+        @humans = Hash(Regex, String).new
+        @acronyms = Hash(String, String).new
+        @acronym_regex = /(?=a)b/
+      end
+
+      def acronym(word)
+        @acronyms[word.downcase] = word
+        @acronym_regex = /#{acronyms.values.join("|")}/
+      end
+
+      def plural(rule : String | Regex, replacement : String)
+        if rule.is_a?(String)
+          @uncountables.delete(rule)
+          rule = /#{rule}/
+        end
+        @uncountables.delete(replacement)
+        new_plural = { rule => replacement }
+        @plurals = new_plural.merge(@plurals)
+      end
+
+      def singular(rule : String | Regex, replacement : String)
+        if rule.is_a?(String)
+          @uncountables.delete(rule)
+          rule = /#{rule}/
+        end
+        @uncountables.delete(replacement)
+        new_singular = { rule => replacement }
+        @singulars = new_singular.merge(@singulars)
+      end
+
+      def irregular(singular : String, plural : String)
+        @uncountables.delete(singular)
+        @uncountables.delete(plural)
+
+        s0 = singular[0]
+        srest = singular[1..-1]
+
+        p0 = plural[0]
+        prest = plural[1..-1]
+
+        if s0.upcase == p0.upcase
+          plural(/(#{s0})#{srest}$/i, "\\1" + prest)
+          plural(/(#{p0})#{prest}$/i, "\\1" + prest)
+
+          singular(/(#{s0})#{srest}$/i, "\\1" + srest)
+          singular(/(#{p0})#{prest}$/i, "\\1" + srest)
+        else
+          plural(/#{s0.upcase}(?i)#{srest}$/,   p0.upcase   + prest)
+          plural(/#{s0.downcase}(?i)#{srest}$/, p0.downcase + prest)
+          plural(/#{p0.upcase}(?i)#{prest}$/,   p0.upcase   + prest)
+          plural(/#{p0.downcase}(?i)#{prest}$/, p0.downcase + prest)
+
+          singular(/#{s0.upcase}(?i)#{srest}$/,   s0.upcase   + srest)
+          singular(/#{s0.downcase}(?i)#{srest}$/, s0.downcase + srest)
+          singular(/#{p0.upcase}(?i)#{prest}$/,   s0.upcase   + srest)
+          singular(/#{p0.downcase}(?i)#{prest}$/, s0.downcase + srest)
+        end
+      end
+
+      def uncountable(*words)
+        @uncountables.add(words.to_a)
+      end
+
+      def human(rule : String | Regex, replacement : String)
+        rule = /#{rule}/ if rule.is_a?(String)
+        @humans = { rule => replacement }.merge(@humans)
+      end
+
+      def clear(scope = :all)
+        scopes = scope == :all ? [:plurals, :singulars, :uncountables, :humans] : [scope]
+
+        scopes.each do |scope|
+          case scope
+          when :plurals
+            @plurals = Hash(Regex, String).new
+          when :singulars
+            @singulars = Hash(Regex, String).new
+          when :uncountables
+            @uncountables = Uncountables.new
+          when :humans
+            @humans = Hash(Regex, String).new
+          end
+        end
+      end
+    end
+
+    @@inflections = Inflections.new
+    def inflections
+      @@inflections
+    end
+  end
+end
```

Comment:
Can you run `crystal tool format`? That will add this missing newline and make sure everything has a consistent style

Reply:
I added a newline since the tool errors on the file - https://github.com/crystal-lang/crystal/issues/5310

This also reformatted other files as well. I don't love that it reformatted hash value assignments but I'm for consistency.